File: GFG/Maths/productPair.c
#include<stdio.h>
#include<stdlib.h>
#include<stdbool.h>
#include<limits.h>
#include<string.h>
/*

# Author			: @RAJ009F
# Topic or Type 	: DSME/DP
# Problem Statement	: find the product of two elements in an arry is equals to a given number 
# Description		: 
# Complexity		: 
=======================
#sample output
----------------------

=======================
*/
void printPairProduct(int arr[], int n, int K)
{
	bool H[K];
	
	memset(H, 0, sizeof(H));
	
	
	int i;
	
	for(i=0;i<n;i++)
	{
		if(K%arr[i]==0 && H[K/arr[i]])
		{
		printf(" pair [(%d * %d)= %d] exists\n", arr[i], K/arr[i], K);
		return;
		}else
			H[arr[i]] = 1;
		
		
		
	}
	
	
	
	
	
}

int main()
{
	int arr[] = {10,30,34,90,40};
	
	printPairProduct(arr, sizeof(arr)/sizeof(arr[0]), 300);
	
	
}

File: GFG/Array/ProductPair.java
import java.io.*;
import java.lang.*;
import java.util.*;

// #!/usr/bin/python -O

// #include <stdio.h>
// #include <stdlib.h>
// #include <string.h>
// #include<stdbool.h>
// #include<limits.h>

// #include<iostream>


// #include<algorithm>
// #include<string>
// #include<map>
// #include<vector>
// using namespace std;

/*

# Author			: @RAJ009F
# Topic or Type 	: GFG/ARRAY
# Problem Statement	: Product Pair 
# Description		: 
# Complexity		: 
=======================
#sample output
----------------------

=======================
*/

class ProductPair
{
	public static void producPair(int arr[],  int n, int x) 
	{
		HashSet<Integer> set =  new HashSet<Integer>();
		
		for(int i=0; i<n; i++)
		{
			if(arr[i] == 0 && x==0)
			{
				System.out.println("x = "+x);
			return;
			}
			if(x%arr[i]==0)
			{if(set.contains(arr[i]))
				System.out.println(arr[i]+" * "+x/arr[i]+" = "+x);
			else
				set.add(x/arr[i]);
			}
			
			
		}
		
		
		
	}
	
	public static void main(String args[])
	{
		int arr[] = {-10, 20, 9, 40};
		
		int x= 800;
		producPair(arr, arr.length, x);
		
	}
	
	
}